compiler/rustc_middle/src/hir/place.rs

```diff
@@ -143,3 +143,29 @@ impl<'tcx> Place<'tcx> {
         if projection_index == 0 { self.base_ty } else { self.projections[projection_index - 1].ty }
     }
 }
+
+/// Return true if the `proj_possible_ancestor` represents an ancestor path
```

Comment:
on second thoughts this might better belong in rustc_middle/src/ty/mod.rs near ty::CapturedPlace